frontends/PyCDE/test/polynomial.py

```diff
@@ -21,6 +21,8 @@ def __init__(self, name: str, coefficients: list[int], **kwargs):
     """coefficients is in 'd' -> 'a' order."""
     self.instanceName = name
     self.coefficients = Parameter(coefficients)
+    self.module_name = Parameter("PolyComputeForCoeff_" +
```

Comment:
Having the user specify a magic parameter named `module_name` seems... magic. None of the other parameters get any special treatment. Is the point of this so the generated modules have non-conflicting symbol names? If so, perhaps the __Module constructor can have some generic logic to stringify and concatenate Parameter values. If the point is for the user to directly control the module name for readability, etc., I think we could expose it a different way than a magic Parameter.

Reply:
Yeah, I was concerned about this as well.

Reply:
The intention is to solve both problems. I considered trying to automatically string-ify the parameters, but I've not had much success in the past getting anything readable with that method.

Reply:
What do you think of the simple method call I switched to? It's more explicit with the same result.

Reply:
The explicit public API to set a module name makes sense, but could that just store a string on `self.module_name`? Seems like it still uses a Parameter under the hood, which isn't the end of the world, but it's still a little confusing that we need to remember to `pop` out that Parameter, if it exists.

Reply:
> generic logic to stringify and concatenate Parameter values

FYI I was curious about this, and I took a look. I'll have a PR in a second... the output is fairly readable IMO.

Reply:
I drafted my thoughts here: https://github.com/llvm/circt/pull/1230. Let me know what you think.

Reply:
In some sense it is a design parameter in that is should be available to the generator and needs to be unique'd on. That it requires a parameter on the generate method is unfortunate....